### PR TITLE
feat: removed kafka, added priority queue manage package

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: confd/production.tmpl [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-        "ssl_cafile": "{{ getv "/configuration/discrete/worker/kafka/ssl/cafile" "null" }}",
-        "ssl_certfile": "{{ getv "/configuration/discrete/worker/kafka/ssl/certfile" "null" }}",
-        "ssl_keyfile": "{{ getv "/configuration/discrete/worker/kafka/ssl/keyfile" "null" }}",
-        "ssl_password": "{{ getv "/configuration/discrete/worker/kafka/ssl/password" "null" }}"
+    "queue": {
+        "job_type": "{{ getv "/configuration/discrete/worker/queue/job/type" "Discrete-Tiling" }}",
+        "task_type": "{{ getv "/configuration/discrete/worker/queue/task/type" "Discrete-Tiling" }}",
+        "job_manager_url": "{{ getv "/configuration/discrete/worker/queue/job/manager/url" "http://127.0.0.1:8082" }}",
+        "heartbeat_manager_url": "{{ getv "/configuration/discrete/worker/queue/heartbeat/manager/url" "http://127.0.0.1:8081" }}",
+        "heartbeat_interval_ms": {{ getv "/configuration/discrete/worker/queue/heartbeat/interval/ms" "0.2" }},
```
Comment (asafMasa) [inline confd/production.tmpl]: seconds

File: confd/production.toml [hunk 2]
```diff
@@ -2,6 +2,13 @@
 src = "production.tmpl"
 dest = "./config/default.json"
 keys = [
+    "/configuration/discrete/worker/update/delay/seconds",
+    "/configuration/discrete/worker/queue/task/type",
+    "/configuration/discrete/worker/queue/job/type",
+    "/configuration/discrete/worker/queue/job/manager/url",
+    "/configuration/discrete/worker/queue/heartbeat/manager/url",
+    "/configuration/discrete/worker/queue/heartbeat/interval/ms",
+    "/configuration/discrete/worker/queue/dequeue/interval/ms",
```
Comment (asafMasa) [inline confd/production.toml]: seconds

File: src/handler.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            while True:
+                task = await self.queue_handler.dequeue(self.__config["queue"]["dequeue_interval_ms"])
+                task_id = task["id"]
+                job_id = task["jobId"]
+                parameters = task["parameters"]
+                is_valid, reason = utilities.validate_data(parameters)
+
+                if not is_valid:
+                    if task_id and job_id:
+                        self.log.error("Validation error - could not process request. Comitting from queue")
```
Comment (asafMasa) [inline src/handler.py:29]: log the job and task id (and discreteID, version)

File: src/handler.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                    else:
+                        self.log.error("Validation error - could not process request and could not save status to DB. Comitting from queue")
```
Comment (asafMasa) [inline src/handler.py:31]: log the job or task id if exists  (and discreteID, version)

File: src/handler.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        self.__worker = Worker()
+        self.queue_handler = TaskHandler(self.__config["queue"]["job_type"], self.__config["queue"]["task_type"],
+                                         self.__config["queue"]["job_manager_url"], self.__config["queue"]["heartbeat_manager_url"],
+                                         self.__config["queue"]["heartbeat_interval_ms"], self.log)
+
+    async def handle_tasks(self):
+        try:
+            while True:
+                task = await self.queue_handler.dequeue(self.__config["queue"]["dequeue_interval_ms"])
+                task_id = task["id"]
```
Comment (asafMasa) [inline src/handler.py:22]: is there a log about the dequeud task before starting to work on it? if not add that log

File: src/worker.py [hunk 6]
```diff
@@ -20,22 +21,20 @@ def vrt_file_location(self, discrete_id):
...
 
-    def buildvrt_utility(self, job_id, task_id, discrete_id, version, zoom_levels):
-        
-        job_data = request_connector.get_job_data(job_id)
+    def buildvrt_utility(self, job_data, zoom_levels):
         if not (job_data["parameters"]["fileNames"] and job_data["parameters"]["originDirectory"]):
-            raise VRTError("jobData didn't have source files data, for jobID: {0} taskID: {1} discreteID: {2}, version: {3} jobData: {4}"
-            .format(job_id, task_id, discrete_id, version, job_data))
+            raise VRTError("jobData didn't have source files data, for jobID: {0} taskID: {1} discreteID: {2}, version: {3}"
+            .format(job_data['jobId'], job_data['id'], job_data['parameters']['discreteId'], job_data['parameters']['version']))
```
Comment (asafMasa) [inline src/worker.py:37]: .format(job_data['jobId'], job_data['id'], job_data['parameters']['discreteId'], job_data['parameters']['version']))
try to put in utils, and use the same format for several logs

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: confd/production.tmpl (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply confd/production.tmpl]: fixed in e6f2d1f

File: confd/production.toml (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply confd/production.toml]: e6f2d1f fixed

File: src/worker.py (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply src/worker.py]: c9b0d53 fixed

File: src/handler.py (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/handler.py]: c9b0d53  fixed

File: src/handler.py (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply src/handler.py:22]: dffa9f2 fixed

File: src/handler.py [hunk 7]
```diff
@@ -64,12 +57,12 @@ async def do_task_loop(self, task_values):
             success, reason = self.execute_task(task_values, zoom_levels)
 
             if success:
-                self.log.info('Successfully finished jobID: {0}, taskID: {1}, discreteID: {2}, version: {3} with zoom-levels: {4}.'
-                              .format(job_id, task_id, discrete_id, version, zoom_levels))
+                self.log.info('Successfully finished {0}'
+                              .format(task_values))
```
Comment (asafMasa) [inline src/handler.py:61]: utilities.task_format_log

File: src/worker.py [hunk 8]
```diff
@@ -21,29 +21,29 @@ def vrt_file_location(self, discrete_id):
...
 
-    def remove_vrt_file(self, job_data, zoom_levels):
+    def remove_vrt_file(self, job_data):
         vrt_path = self.vrt_file_location(job_data['parameters']['discreteId'])
-        self.log.info('Removing vrt file from path "{0}", jobID: {1}, taskID: {2}, discreteID {3}, with zoom-levels {4}'.format(vrt_path, job_data['jobId'], job_data['id'], job_data['parameters']['discreteId'], zoom_levels))
+        self.log.info('Removing vrt file from path "{0}", {1}'.format(vrt_path, utilities.task_format_log(job_data)))
         remove_file(vrt_path)
 
     def remove_s3_temp_files(self, job_data, zoom_levels):
         tiles_location = '{0}/{1}'.format(self.tiles_folder_location, job_data['parameters']['discreteId'])
```
Comment (asafMasa) [inline src/worker.py:30]: remove zoom_levels

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: src/handler.py [hunk 9]
```diff
@@ -50,8 +45,6 @@ async def do_task_loop(self, task_values):
         parameters = task_values["parameters"]
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/handler.py]: fixed in 
5b661e0

File: src/handler.py (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply src/handler.py]: fixed in 
1246757

Comment (CL-SHLOMIKONCHA) [reply src/handler.py]: & 94a6289

File: src/handler.py [hunk 10]
```diff
@@ -13,35 +12,31 @@ def __init__(self):
...
-                task = await self.queue_handler.dequeue(self.__config["queue"]["dequeue_interval_ms"])
+                task = await self.queue_handler.dequeue(self.__config["queue"]["dequeue_interval_seconds"])
+                self.log.info("Task received - {0}".format(task))
                 task_id = task["id"]
                 job_id = task["jobId"]
                 parameters = task["parameters"]
                 is_valid, reason = utilities.validate_data(parameters)
 
                 if not is_valid:
                     if task_id and job_id:
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/handler.py]: removed in 
1246757

File: src/utilities.py [hunk 11]
```diff
@@ -55,8 +55,12 @@ def validate_data(task_values):
...
 
-    if (task_values['minZoomLevel'] > task_values['maxZoomLevel']):
+    if task_values['minZoomLevel'] > task_values['maxZoomLevel']:
         reason = 'Minimum zoom level cannot be greater than maximum zoom level'
         return False, reason
     
     return True, ""
+
+def task_format_log(task_values):
+    return "jobId: {0}, taskId: {1}, discreteID: {2}, version: {3}"\
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/utilities.py]: fixed in 
1246757

File: src/worker.py (hunk 8 above)
Comment (CL-SHLOMIKONCHA) [reply src/worker.py]: removed

**Review by asafMasa — APPROVED** [review]

**Pull request merged**